### DISCUSSION TITLE
actions/upload-artifact : Computed hash does not match with url signature

### DISCUSSION THREAD

**Discussion by jy95** [community/community#27001] [Actions]
<p>Hello,</p>
<p>Using actions/upload-artifact, I managed to upload a file using a customizable env variable : </p>
<pre><code>    env:
      FINAL_FILENAME: Source Code - un catalogue open source d’exercices informatiques.pdf
      LATEX_FILENAME: main.pdf
      MAIN_LATEX: main.tex
    steps:
       .....
      - name: rename ${{ env.LATEX_FILENAME }} to ${{ env.FINAL_FILENAME }}
        run: |
          mv "${{ env.LATEX_FILENAME }}" "${{ env.FINAL_FILENAME }}"
      - name: Upload result file
        uses: actions/upload-artifact@v2
        with:
          name: ${{ env.FINAL_FILENAME }}
          path: ${{ env.FINAL_FILENAME }}
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/62976fc1103177d3787f7f812eb16917500bc675.png" data-download-href="/uploads/short-url/e4bf5BzJiVvOX24RfIItjcAzFwF.png?dl=1" title="bug42.PNG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/9306961/181101114-bac03b84-bcd4-4de8-8fb8-f26e4da118fe.png" alt="bug42.PNG" data-base62-sha1="e4bf5BzJiVvOX24RfIItjcAzFwF" width="690" height="257" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/62976fc1103177d3787f7f812eb16917500bc675_2_690x257.png, https://aws1.discourse-cdn.com/github/optimized/2X/6/62976fc1103177d3787f7f812eb16917500bc675_2_1035x385.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/6/62976fc1103177d3787f7f812eb16917500bc675_2_1380x514.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/62976fc1103177d3787f7f812eb16917500bc675_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">bug42.PNG</span><span class="informations">1889×704 41.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Which gives me the following url : </p>
<p><a href="https://github.com/jy95/M-moire---Source-Code-un-catalogue-open-source-d-exercices-informatiques/suites/664591803/artifacts/5896280" rel="nofollow noopener">https://github.com/jy95/M-moire—Source-Code-un-catalogue-open-source-d-exercices-informatiques/suites/664591803/artifacts/5896280</a> </p>
<p>When I request the url, I got this </p>
<pre><code>{"count":47,"value":"Computed hash does not match with url signature"}
</code></pre>
<p>Any ideas ?<br>
PS : Please avoid answers like “use a filename without spaces” as I know the action works with regular filenames</p>

**Comment by jy95** [comment]
<p>As far that I know, the filename doesn’t contain actions/toolkit <a>not supported characters</a> (and thus is correctly uploaded so something seems wrong with Github REST API …)</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/jy95">@jy95</a> ,</p>
<p>I can reproduce the same on my side. The error is due to quote(highlighted)</p>
<pre><code>FINAL_FILENAME: Source Code - un catalogue open source d&lt;font color="#FF0000"&gt;’&lt;/font&gt;exercices informatiques.pdf
</code></pre>
<p>is used in your file name, it’s not supported.</p>
<p>Can use single quote ’ instead.</p>
<pre><code>FINAL_FILENAME: Source Code - un catalogue open source d&lt;font color="#FF0000"&gt;'&lt;/font&gt;exercices informatiques.pdf
</code></pre>
<p><img width="471" height="153" alt="artifactname.png" src="https://user-images.githubusercontent.com/52772761/181101115-03b17376-31d2-4205-8e05-27547c577a8a.png"></p>
<p>Thanks.</p>

**Comment by jy95** [comment]
<p>Thanks for the answer : it seems Github team never tried this character ^^</p>
